### PR TITLE
feat(1082): [4] implement getBranchList function

### PR DESCRIPTION
**Pull request by yuichi10**
## Context
To get pipelines, we need a function which get list of branch.

## Objective
Implement getBranchList function.

## References
[octkit/rest api](https://octokit.github.io/rest.js/#api-Repos-getBranches)

Blocked by 
- https://github.com/screwdriver-cd/data-schema/pull/240
- https://github.com/screwdriver-cd/scm-base/pull/57

Related to screwdriver-cd/screwdriver#1082

### CODE REVIEW

File: test/index.test.js [hunk 1]
```diff
@@ -1788,4 +1789,62 @@ jobs:
...
+        beforeEach(() => {
+            githubMock.repos.getById.yieldsAsync(null, {
+                full_name: 'dolores/violentdelights'
+            });
+            githubMock.repos.getBranches.yieldsAsync(null, [{
+                name: 'master'
+            }]);
+        });
+
+        it('get branches', () => {
```
[the remark this answers is not in the document]
Comment (tk3fftk) [reply test/index.test.js]: This test case also need `done` and `catch` for a correct test.

File: index.js [hunk 2]
```diff
@@ -915,6 +916,66 @@ class GithubScm extends Scm {
             return false;
         }
     }
+
+    /**
+     * Look up a branches from a repo
+     * @async  _findBranches
+     * @param  {Config}     config
```
[the remark this answers is not in the document]
Comment (yuichi10) [reply index.js]: You are right. I will fix

File: index.js [hunk 3]
```diff
@@ -915,6 +916,66 @@ class GithubScm extends Scm {
...
+        if (branches.length === BRANCH_PAGE_SIZE) {
+            config.page += 1;
+            const nextPageBranches = await this._findBranches(config);
+
+            branches = branches.concat(nextPageBranches);
+        }
+
+        const branchInfos = [];
+
+        branches.forEach((branch) => {
```
[the remark this answers is not in the document]
Comment (yuichi10) [reply index.js]: Definitely this way is better! Thank you!

File: test/index.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (yuichi10) [reply test/index.test.js]: Thank you I will fix.

File: test/index.test.js [hunk 4]
```diff
@@ -1788,4 +1789,62 @@ jobs:
...
+                assert.calledWith(githubMock.repos.getBranches, {
+                    owner: 'dolores',
+                    repo: 'violentdelights',
+                    page: 1,
+                    per_page: 100
+                });
+            });
+        });
+
+        it('get a lot of branches', (done) => {
```
[the remark this answers is not in the document]
Comment (yuichi10) [reply test/index.test.js]: I will fix.

File: test/index.test.js [hunk 5]
```diff
@@ -1788,4 +1789,63 @@ jobs:
...
+            scmUri: 'github.com:1289:branchName',
+            token: 'fakeToken'
+        };
+
+        beforeEach(() => {
+            githubMock.repos.getById.yieldsAsync(null, {
+                full_name: 'dolores/violentdelights'
+            });
+            githubMock.repos.getBranches.yieldsAsync(null, [{
+                name: 'master'
```
[the remark this answers is not in the document]
Comment (yuichi10) [reply test/index.test.js]: I added more data to mock!

[the remark this answers is not in the document]
Comment (tkyi) [reply test/index.test.js]: Oh wow that is a lot more data!

**Pull request merged**